Add dependecy checking for programs needed to download the driver

mauritiusdadd opened the pull request:
Regarding issue #41, I added a simple dependency checking to the Makefile for the firmware extraction: now when you run make, before performing the actual download, the presence of the following programs (which are needed to download the driver) is checked: `curl`, `xzcat`, `cpio`.

hobarrera commented:
Commit message [should be shorter than 50 messages](https://git-scm.com/docs/git-commit#_discussion) and this one looks just too verbose.

Code-wise this looks perfect and a great addition.

mauritiusdadd commented:
@hobarrera : 
>Commit message should be shorter than 50 messages and this one looks just too verbose.

Oh, I didn't know about that, thank you for pointing it out!

hobarrera commented:
👍

mauritiusdadd commented:
I added the new hashes, obtained from #41, to the extraction script. For testing purpose, I also added an option to ignore the checksum and check only the firmware header and footer (but it is not done by default, you need to explicitly pass the `-i` option to the extraction script).

patjak commented:
Merged, thanks!
